src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -135,9 +139,50 @@ public Optional<Address> getRetiringFederationAddress() {
         return federatorSupport.getRetiringFederationAddress();
     }
 
+    @Override
+    public Optional<Federation> getProposedFederation() {
+        if (!federatorSupport.getConfigForBestBlock().isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+       
+        Optional<Address> proposedFederationAddress = getProposedFederationAddress();
+        Integer federationSize = federatorSupport.getProposedFederationSize()
+            .orElse(FEDERATION_NON_EXISTENT.getCode());
+        if (federationSize == FEDERATION_NON_EXISTENT.getCode() || proposedFederationAddress.isEmpty()) {
+            return Optional.empty();
+        }
+
+        List<FederationMember> members = IntStream.range(0, federationSize)
+            .mapToObj(i -> new FederationMember(
+                federatorSupport.getProposedFederatorPublicKeyOfType(i, KeyType.BTC)
+                    .map(ECKey::getPubKey)
+                    .map(BtcECKey::fromPublicOnly)
+                    .orElse(null),
+                federatorSupport.getProposedFederatorPublicKeyOfType(i, KeyType.RSK)
+                    .orElse(null),
+                federatorSupport.getProposedFederatorPublicKeyOfType(i, KeyType.MST)
+                    .orElse(null)
+            ))
+            .toList();
+
+        FederationArgs federationArgs = new FederationArgs(
+            members,
+            federatorSupport.getProposedFederationCreationTime().orElse(null),
+            federatorSupport.getProposedFederationCreationBlockNumber().orElse(null),
+            federatorSupport.getBtcParams()
+        );
+
+        Federation initialFederation = FederationFactory.buildStandardMultiSigFederation(federationArgs);
+        Federation expectedFederation = getExpectedFederation(initialFederation, proposedFederationAddress.get());
```

Comment:
I see how `getExpectedFederation` builds the federation, and I think this method is unsuitable for this case. At the beginning, you have already checked if `RSKIP419` is active, so we can be sure that the proposed federation will be a `P2shFederation`. if not, I would like to understand why you would be a different type?

Reply:
Thanks for pointing this out :) , I have made the changes. Let me know if this looks good now.